test/Preact10ShallowDiff/Preact10ShallowDiff_test.tsx

```diff
@@ -0,0 +1,1709 @@
+/**
```

Comment:
I wouldn't worry too much about reviewing this file in detail (unless you want to!). It's just a copy and slightly modified to be TypeScript and Preact friendly of this test in react-shallow-renderer: https://github.com/enzymejs/react-shallow-renderer/blob/802c735ee53bf2d965797760698cacbd46088f66/src/__tests__/ReactShallowRenderer-test.js